code/modules/clothing/spacesuits/rig.dm

```diff
@@ -85,6 +92,66 @@
 	allowed = list(/obj/item/device/flashlight,/obj/item/weapon/tank,/obj/item/weapon/storage/bag/ore,/obj/item/device/t_scanner,/obj/item/weapon/pickaxe, /obj/item/device/rcd, /obj/item/weapon/wrench/socket)
 	max_heat_protection_temperature = SPACE_SUIT_MAX_HEAT_PROTECTION_TEMPERATURE
 	pressure_resistance = 200 * ONE_ATMOSPHERE
+	var/obj/item/clothing/head/helmet/space/rig/H = null
+	var/head_type = /obj/item/clothing/head/helmet/space/rig
+	var/obj/item/weapon/cell/cell = null
+	var/cell_type = /obj/item/weapon/cell/high //The cell_type we're actually using
+	var/list/modules = list()
+	actions_types = list(/datum/action/item_action/toggle_rig_helmet)
+
+/obj/item/clothing/suit/space/rig/New()
+	..()
+	cell = new cell_type
+	H = new head_type
+	H.rig = src
+
+/obj/item/clothing/suit/space/rig/Destroy()
+	qdel(cell)
+	cell = null
+	if(H.loc == src || !H.loc)
+		qdel(H)
+	H = null
+	..()
+
+/obj/item/clothing/suit/space/rig/unequipped(mob/living/carbon/human/user, var/from_slot = null)
+	..()
+	if(from_slot == slot_wear_suit && istype(user))
+		deactivate_suit(user)
+
+/obj/item/clothing/suit/space/rig/proc/toggle_helmet(mob/living/carbon/human/user)
+	if(H)
+		if(!user.head)
+			to_chat(user, "<span class = 'notice'>\The [H] extends from \the [src].</span>")
+			user.equip_to_slot(H, slot_head)
+			H = null
+			initialize_suit(user)
+	else
+		if(user.head && istype(user.head, head_type))
+			var/obj/I = user.head
+			to_chat(user, "<span class = 'notice'\The [I] retracts into \the [src].</span>")
+			user.u_equip(I,0)
+			I.forceMove(src)
+			H = I
+			deactivate_suit(user)
+
+
+/obj/item/clothing/suit/space/rig/proc/initialize_suit(mob/user)
+	for(var/obj/item/rig_module/R in modules)
+		R.activate(user,src)
+
+/obj/item/clothing/suit/space/rig/proc/deactivate_suit(mob/user)
+	for(var/obj/item/rig_module/R in modules)
+		R.deactivate(user,src)
+
+/obj/item/clothing/suit/space/rig/attackby(obj/W, mob/user)
+	if(!H && istype(W, head_type))
+		to_chat(user, "<span class = 'notice'>You attach \the [W] to \the [src].</span>")
+		user.drop_item(W, force_drop = 1)
```

Comment:
you should check the return value of this before proceeding, the thing may be glued to your hand. you can also specify `src` as the second parameter to have it `forceMove` the thing for you.